Added option to select snapshot filename in scripts.

NissimHadar opened the pull request:
# Description
The Window,takeSnapshot() and Window.takeSecondarySnapshot() scripting commands now accept an optional filename.  If this filename is present then the snapshot will be saved as `<filename>.jpg`.
# Tests
Work In Progress
# User Documentation
Work In Progress

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6933/app_12343_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6933/HighFidelity-Beta-PR12343-cf87052b7fd852c1446931f7a0433848b3d5edb6.exe

samcake commented on interface/src/ui/Snapshot.cpp:
```diff
@@ -104,7 +104,15 @@ QFile* Snapshot::savedFileForSnapshot(QImage & shot, bool isTemporary) {
+	// 'jpg" is appended, as the image is saved in jpg format.  This is the case for all snapshots
+	//       (see definition of FILENAME_PATH_FORMAT)
+    QString filename;
+    if (userSelectedFilename != "") {
```
Is it the recommended way to compare strings ?
isn;t there a method on qstring for that to compare against an empty string ?

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6946/app_12343_PR.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6946/HighFidelity-Beta-PR12343-faec9a62b6320a5ad05c1cb660871ed9881ff20e.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6946/HighFidelity-Beta-PR12343-faec9a62b6320a5ad05c1cb660871ed9881ff20e.exe

NissimHadar commented:
Fixed.  The recommended way is not with an empty sting but with a null
string.  The default QString constructor creates a null string (not the
same thing).  I also changed the default parameter to match.

On Tue, Feb 6, 2018 at 10:34 AM, Sam Gateau <notifications@github.com>
wrote:

> *@samcake* commented on this pull request.
> ------------------------------
>
> In interface/src/ui/Snapshot.cpp
> <https://github.com/highfidelity/hifi/pull/12343#discussion_r166399895>:
>
> > @@ -104,7 +104,15 @@ QFile* Snapshot::savedFileForSnapshot(QImage & shot, bool isTemporary) {
>
>      QDateTime now = QDateTime::currentDateTime();
>
> -    QString filename = FILENAME_PATH_FORMAT.arg(username, now.toString(DATETIME_FORMAT));
> +    // If user has requested specific filename then use it, else create the filename
> +	// 'jpg" is appended, as the image is saved in jpg format.  This is the case for all snapshots
> +	//       (see definition of FILENAME_PATH_FORMAT)
> +    QString filename;
> +    if (userSelectedFilename != "") {
>
> Is it the recommended way to compare strings ?
> isn;t there a method on qstring for that to compare against an empty
> string ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/highfidelity/hifi/pull/12343#pullrequestreview-94455904>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Abrm6gDdgLfmZHPiFIrC6H-xiPbx0vLuks5tSJsugaJpZM4R6XTK>
> .
>

ctrlaltdavid commented:
The JSDoc comments in WindowScriptingInterface.h need updating to include the added parameters.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6978/app_12343_PR.apk

NissimHadar commented:
Please build me

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6981/app_12343_PR.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6981/HighFidelity-Beta-PR12343-5579eac034a9ca5cd1ed13032a1c9e1f1b538a29.exe

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6981/HighFidelity-Beta-PR12343-5579eac034a9ca5cd1ed13032a1c9e1f1b538a29.dmg

sbennett77 commented:
Currently Testing.

sbennett77 commented:
Passed.

@conklin94122
